src/Compilers/CSharp/Test/Semantic/Diagnostics/GetDiagnosticsTests.cs

```diff
@@ -394,5 +394,43 @@ public void CompilingCodeWithSameParseOptionsInMultipleSyntaxTreesShouldReportOn
             Assert.True(diagnostics[0].Location.SourceTree.Equals(syntaxTree1));
             Assert.True(diagnostics[1].Location.SourceTree.Equals(syntaxTree2));
         }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingNullKeyFile()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: null);
+            CreateCompilation(string.Empty, options: options).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingEmptyKeyFile()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: string.Empty);
+            CreateCompilation(string.Empty, options: options).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingNullKeyFile_PublicSign()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: null, publicSign: true);
+            CreateCompilation(string.Empty, options: options).VerifyDiagnostics(
+                // error CS8102: Public signing was specified and requires a public key, but no public key was specified.
+                Diagnostic(ErrorCode.ERR_PublicSignButNoKey).WithLocation(1, 1));
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingEmptyKeyFile_PublicSigne()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: string.Empty, publicSign: true);
+            CreateCompilation(string.Empty, options: options).VerifyDiagnostics(
+                // error CS8106: Option 'CryptoKeyFile' must be an absolute path.
```

Comment:
Null here means uninitialized/not set. An empty string means that the user specifically set it to something, and we will validate it. Since an empty string is not a valid path, we will error out.
What do you propose to do instead?

---
In reply to: [109722102](https://github.com/dotnet/roslyn/pull/18407#discussion_r109722102) [](ancestors = 109722102)